python/lsst/jointcal/jointcal.py

```diff
@@ -656,25 +710,42 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
                 config=self.config.photometryRefObjLoader,
                 log=self.log)
         outputs = self.run(**inputs, tract=tract)
+        self._put_metrics(butlerQC, outputs.job, outputRefs)
         if self.config.doAstrometry:
             self._put_output(butlerQC, outputs.outputWcs, outputRefs.outputWcs,
                              inputs['inputCamera'], "setWcs")
         if self.config.doPhotometry:
             self._put_output(butlerQC, outputs.outputPhotoCalib, outputRefs.outputPhotoCalib,
                              inputs['inputCamera'], "setPhotoCalib")
 
+    def _put_metrics(self, butlerQC, job, outputRefs):
+        """Persist all measured metrics stored in a job.
+
+        Parameters
+        ----------
+        butlerQC : `lsst.pipe.base.ButlerQuantumContext`
+            A butler which is specialized to operate in the context of a
+            `lsst.daf.butler.Quantum`; This is the input to `runQuantum`.
+        job : `lsst.verify.job.Job`
+            Measurements of metrics to persist.
+        outputRefs : `list` [`lsst.pipe.base.connectionTypes.OutputQuantizedConnection`]
+            The DatasetRefs to persist the data to.
+        """
+        for key in job.measurements.keys():
+            butlerQC.put(job.measurements[key], getattr(outputRefs, key.fqn.replace('jointcal.', '')))
```

Comment:
I'm a little surprised this is an attribute on outputRefs rather than an index into outputRefs, but I trust you tested it.

Reply:
`OutputQuantizedConnection` (outputRefs) is a `SimpleNamespace`, which behaves like a `namedtuple`, not a `dict` (even though it has a `keys()` method).